### PR TITLE
Migrate from tsdx to use rollup and jest directly for msal-node

### PR DESCRIPTION
**Pull request by hectormmg**
This PR:
- Removes `tsdx` from `msal-node` dependencies
- Replaces `msal-node` tsdx build and test scripts with their rollup and jest equivalents
- Adds `rollup.config.js` to `msal-node`
- Removes Node v14 from the MSAL Node test matrix
- Uses jest requireActual to properly load symlinked msal-common

### CODE REVIEW

File: lib/msal-node/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        plugins: [
+            typescript( {
+                typescript: require("typescript"),
+                tsconfig: "tsconfig.build.json"
+            }),
+            nodeResolve()
+        ]
+    },
+    {
+        // for cjs build
```
Comment (hectormmg) [inline lib/msal-node/rollup.config.js]:
```suggestion
        // for esm build
```

File: lib/msal-node/src/client/ClientApplication.ts [hunk 2]
```diff
@@ -163,7 +163,7 @@ export abstract class ClientApplication {
             if (e instanceof AuthError) {
                 e.setCorrelationId(validRequest.correlationId);
             }
-            serverTelemetryManager.cacheFailedRequest(e);
+            serverTelemetryManager.cacheFailedRequest(e as AuthError);
```
Comment (konstantin-msft) [inline lib/msal-node/src/client/ClientApplication.ts]: `serverTelemetryManager.cacheFailedRequest()` accepts any types of error. You don't need to cast it explicitly https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/msal-browser-v3/lib/msal-common/src/telemetry/server/ServerTelemetryManager.ts#L82.

**Review by konstantin-msft — APPROVED** [review]

File: lib/msal-node/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@azure/msal-node",
-  "version": "1.15.0",
+  "version": "1.16.0",
```
Comment (tnorling) [inline lib/msal-node/package.json]: Why was this bumped?

File: lib/msal-node/test/client/PublicClientApplication.spec.ts [hunk 4]
```diff
@@ -524,7 +524,7 @@ describe('PublicClientApplication', () => {
         const logger = new Logger({
             loggerCallback: (level, message, containsPii) => {
                 expect(message).toContain("Message");
-                expect(message).toContain(LogLevel.Info);
+                expect(message).toContain("Info");
```
Comment (tnorling) [inline lib/msal-node/test/client/PublicClientApplication.spec.ts]: Why was this changed?

Comment (hectormmg) [reply lib/msal-node/test/client/PublicClientApplication.spec.ts]: I don't know why it was passing before, but under current config LogLevel.Info evaluates to the number 2 corresponding to its enum index, not to a string. It's not present in the log string. Unless I'm missing something, this is a fix.

Comment (tnorling) [reply lib/msal-node/test/client/PublicClientApplication.spec.ts]: Ah interesting. Thanks!

**Review by tnorling — APPROVED** [review]

File: lib/msal-node/src/client/ClientApplication.ts (hunk 2 above)
Comment (hectormmg) [reply lib/msal-node/src/client/ClientApplication.ts]: Thanks, updated these to stop type casting unsafely.

File: lib/msal-node/package.json (hunk 3 above)
Comment (hectormmg) [reply lib/msal-node/package.json]: Undid the change, no idea how it got bumped.

**Review by sameerag — APPROVED** [review]

File: lib/msal-node/test/client/PublicClientApplication.spec.ts [hunk 5]
```diff
@@ -524,7 +526,7 @@ describe('PublicClientApplication', () => {
         const logger = new Logger({
             loggerCallback: (level, message, containsPii) => {
                 expect(message).toContain("Message");
-                expect(message).toContain(LogLevel.Info);
+                expect(message).toContain(LogLevel.Info.toString());
```
Comment (Robbie-Microsoft) [inline lib/msal-node/test/client/PublicClientApplication.spec.ts:529]: Why is this needed here, but not on the next line?

Comment (hectormmg) [reply lib/msal-node/test/client/PublicClientApplication.spec.ts:529]: Because the level is a number. These two lines should never have passed at the same time.
